spec/puppet/resource_api_spec.rb

```diff
@@ -404,56 +414,124 @@ def set(_context, changes)
         end
       end
     end
+    let(:log_sink) { [] }
 
     before(:each) do
       stub_const('Puppet::Provider::Canonicalizer', Module.new)
       stub_const('Puppet::Provider::Canonicalizer::Canonicalizer', provider_class)
+
+      Puppet::Util::Log.newdestination(Puppet::Test::LogCollector.new(log_sink))
+    end
+
+    after(:each) do
+      log_sink.clear
```

Comment:
due to rspec's semantics, the code gets a new `let(:log_sink)` instance for each example (`it`), so no need to clean up `log_sink`.

What I do wonder though is whether `Puppet::Util::Log.newdestination` will be happy with all the `LogCollector`s accumulating? With https://github.com/puppetlabs/puppet-resource_api/pull/29/files#diff-93830fa29d616f7c87903d08b5b1b29aR34 we have https://github.com/puppetlabs/puppetlabs_spec_helper/blob/4271c7222e3366c57863bc8a77ccc117f5d6ec7d/lib/puppetlabs_spec_helper/puppet_spec_helper.rb#L134-L160 available, which could supersede this entire section here.